### DISCUSSION TITLE
Is Brain.js abandoned? I want to help.

### DISCUSSION THREAD

**Discussion by JeffMII** [BrainJS/brain.js#777] [General]
Brain.js is such an elegant neural network package. It is missing some key features like GPU utilization and convolutional neural networking. The features seem to be in development, but the last activity in them seems to have been early last year.

I would hate to see Brain.js become another abandoned JavaScript project. If help is needed, I can try throwing myself into it. I don't have much experience with developing neural networks yet, but I'm willing to put in the effort if it means Brain.js will become what it can be.

Is there any development information available that lists what has been done and what needs to be done?

**Comment by JeffMII** [comment]
Ok. So, it looks like there was a push to the master branch just an hour ago so I just missed it as I was searching through the branches for activity.

**Comment by robertleeplummerjr** [comment]
It is active, it was just slow going in the migration over to typescript.  Now that is completed.  Volunteering is welcomed!

**Reply by JeffMII** [reply]
Understood. I'm glad the project is still active. Creating the type files for everything can be tedious, or if you refactor everything to TypeScript, that's probably even more tedious.

From what I've learned so far, I prefer Brain.js over Tensorflow.js. It's more elegant in my opinion.

I won't promise I will be able to do much, but I'd like to try. I'm still learning neural networks so this could be a meaningful way to do that. Do you have any pointers to help me start? If not that's fine. I can sift through the code and docs to figure it out.
